static/js/disaster_dashboard/data_fetcher.ts

```diff
@@ -0,0 +1,209 @@
+    }
+    const decrementByYear =
+      new Date(maxDate).getFullYear() - new Date(minDate).getFullYear() >
+      MAX_YEARS;
+    const dateList = [];
+    const currDate = new Date(maxDate);
+    const endDate = new Date(minDate);
+    const dateStringCut = decrementByYear ? 4 : 7;
+    while (currDate >= endDate) {
+      // currDate is in local time but toISOString will get the iso string of the
+      // date in UTC. So first we need to convert currDate to UTC date without
+      // changing the date.
+      const utcDate = new Date(currDate.toLocaleDateString() + " UTC");
+      dateList.push(utcDate.toISOString().substring(0, dateStringCut));
+      if (decrementByYear) {
+        currDate.setFullYear(currDate.getFullYear() - 1);
+      } else {
+        currDate.setMonth(currDate.getMonth() - 1);
+      }
+    }
+    return dateList;
+  });
+}
+
+/**
+ * Get promise for event points for a specific event type, place, and date
+ * @param eventType event type to get data for
+ * @param place place to get data for
+ * @param date date to get data for (YYYY-MM)
+ * @param disasterType disaster type that the event type belongs to
+ */
+function fetchEventTypeData(
+  eventType: string,
+  place: string,
+  date: string,
+  disasterType: DisasterType
+): Promise<DisasterEventPoint[]> {
+  return axios
+    .get<DisasterApiEventPoint[]>("/api/disaster-dashboard/data", {
+      params: {
+        eventType: eventType,
+        place: place,
+        date: date,
+      },
+    })
+    .then((resp) => {
+      const result = [];
+      resp.data.forEach((eventData) => {
+        const intensity = {};
+        if (disasterType in DISASTER_EVENT_INTENSITIES) {
+          for (const prop of DISASTER_EVENT_INTENSITIES[disasterType]) {
+            if (prop in eventData) {
+              intensity[prop] = eventData[prop];
+            }
+          }
+        }
+        result.push({
+          placeDcid: eventData.eventId,
+          placeName: eventData.name || eventData.eventId,
+          latitude: eventData.latitude,
+          longitude: eventData.longitude,
+          disasterType,
+          startDate: eventData.startDate,
+          intensity,
+          endDate: eventData.endDate,
+        });
+      });
+      return result;
+    });
+}
+
+/**
+ * Get promise for all event points for a specific disaster type, place, and date
+ * @param disasterType disaster type to get data for
+ * @param place place to get data for
+ * @param date date to get data for (can be YYYY or YYYY-MM)
+ */
+export function fetchDisasterData(
+  disasterType: DisasterType,
+  place: string,
+  date: string
+): Promise<DisasterEventPoint[]> {
+  // Dates to fetch data for. If date argument is a year, we want to fetch data
+  // for every month in that year.
+  const dates = [];
+  if (date.length === 4) {
+    for (let i = 1; i < 13; i++) {
+      dates.push(`${date}-${i < 10 ? "0" : ""}${i}`);
+    }
+  } else if (date.length === 7) {
+    dates.push(date);
+  }
```

Comment:
should we print to console if we're skipping disasters with YYYY-MM-DD format?

Reply:
I don't think that's necessary because the date we're getting data for will either be YYYY or YYYY-MM, which is specified in the function comment

Reply:
i see - i missed that. could you call this out more in the function comment?